### PR TITLE
fix compilation under 32-bit

### PR DESCRIPTION
**Pull request by neheb**
size_t is an unsigned int under 32-bit, causing a redeclaration error.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by KarlStraussberger** [comment]
I worked out a way to use templates which is more generic

**Comment by neheb** [comment]
Updated. Please review. I had to remove 
template <> as that does not work with a declaration.

File: src/web/pages.h [hunk 1]
```diff
@@ -198,11 +198,10 @@ class configLoad : public WebRequestHandler {
     std::vector<ConfigValue> dbEntries;
     std::map<std::string, pugi::xml_node*> allItems;
     void createItem(pugi::xml_node& item, const std::string& name, config_option_t id, config_option_t aid);
-    static void setValue(pugi::xml_node& item, bool value);
+    template <typename T>
+    static void setValue(pugi::xml_node& item, const T value);
```
Comment (KarlStraussberger) [inline src/web/pages.h:202]: You may even make it a rereference 
```
    template <typename T>
    static void setValue(pugi::xml_node& item, const T& value);
```

File: src/web/pages.h [hunk 2]
```diff
@@ -198,11 +198,10 @@ class configLoad : public WebRequestHandler {
...
     static void setValue(pugi::xml_node& item, const std::string& value);
-    static void setValue(pugi::xml_node& item, int value);
-    static void setValue(pugi::xml_node& item, unsigned int value);
-    static void setValue(pugi::xml_node& item, size_t value);
+    static void setValue(pugi::xml_node& item, const fs::path value);
```
Comment (KarlStraussberger) [inline src/web/pages.h:204]: You don't have to declare this here. It works as a template override with `template<>` before the definition as a posted

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]

File: src/web/pages.h (hunk 1 above)
Comment (neheb) [reply src/web/pages.h]: Done.

File: src/web/pages.h (hunk 2 above)
Comment (neheb) [reply src/web/pages.h]: Removed.

**Comment by neheb** [comment]
I think this is what you were looking for.

**Pull request merged**